File: stacks&queues/Stack.java
package linked_lists;

public class Stack {
	Node head;
	int size;
	
		
	public void push(int value) {
		Node node = new Node(value);
		
		if(this.head != null) {
			node.next = this.head;	
		}
		
		this.head = node;
		this.size++;
	}
	
	public int pop() {
				
		int popped = this.head.data;
		this.head = this.head.next;
		this.size--;
		return popped;
	}
	
	public boolean isEmpty() {
		if(this.size == 0) {
			return true;
		}else {
			return false;
		}
	}
	
	public int size() {
		if(isEmpty()) {
			return 0;
		}else {
			return this.size;
		}
		
	}
	
	public int peek() {
		if(isEmpty()) {
			return 0;
		}else {
			return this.head.data;
		}
	}
	
	public void printStack() {
		Node currentNode = this.head;
		
		while(currentNode != null) {
			System.out.print(currentNode.data+" -> ");
			currentNode = currentNode.next;
		}
		
		System.out.println("end");
	}
	
	class Node{
		int data;
		Node next;
		
		Node(int data){
			this.data = data;
			this.next = null;
		}
	}
	
	public static void main(String[] args) {
		
		Stack stack1 = new Stack();
		
		stack1.push(4);
		stack1.push(9);
		stack1.push(6);
		stack1.push(1);

		stack1.printStack();
		
		System.out.println(stack1.size());
		System.out.println(stack1.pop());
		System.out.println(stack1.isEmpty());
		System.out.println(stack1.size());
		System.out.println(stack1.peek());
		
		stack1.printStack();
		
		
	}
	
	
}

File: stacks&queues/stack.c
#include<stdio.h>
#include<stdlib.h>
#include<stdbool.h>

typedef struct Node{
    int data;
    struct Node* next;
}Node;


typedef struct Stack{
    struct Node* head;
    int size;
}Stack;


Stack* createStack(){
    Stack* newStack = (Stack*)malloc(sizeof(Stack));

    if(!newStack){
        printf("Memory allocation failed!!\n");
    }

    newStack->head = NULL;
    newStack->size = 0;

    return newStack;
}

void push(Stack *s,int value){
    Node* newNode = (Node*)malloc(sizeof(Node));

    if(!newNode){
        printf("Memory allocation failed!!\n");
    }


    newNode->data = value;

    if(s->head != NULL){
        newNode->next = s->head;
    }

    s->head = newNode;
    s->size += 1;

}

int pop(Stack* s){

    if(s->size == 0){
        return NULL;
    }

    int popped = s->head->data;
    s->head = s->head->next;
    s->size -= 1;
    return popped;
}

bool isEmpty(Stack* s){
    if(s->size == 0){
        return true;
    }else{
        return false;
    }
}

int size(Stack* s){
    return s->size;
}

int peek(Stack* s){
    if(s->size == 0){
        return NULL;
    }
    return s->head->data;
}

int main(){
    Stack* stack1 = createStack();

    push(stack1, 3);
    push(stack1, 9);
    push(stack1, 7);
    push(stack1, 1);

    printf("%d\n",size(stack1));
    printf("%d\n",pop(stack1));
    printf("%d\n",isEmpty(stack1));
    printf("%d\n",size(stack1));
    printf("%d\n",peek(stack1));

    //Free memories


    return 0;
}
